server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java

```diff
@@ -0,0 +1,666 @@
+                }
+            }
+
+            // Visit positions, offsets, and payloads
+            {
+                TermsEnum termsEnum = terms.iterator();
+                while (termsEnum.next() != null) {
+                    cancellationChecker.logEvent();
+                    termsEnum.docFreq();
+                    termsEnum.totalTermFreq();
+                    postings = termsEnum.postings(postings, PostingsEnum.ALL);
+                    while (postings.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+                        if (terms.hasPositions()) {
+                            for (int pos = 0; pos < postings.freq(); pos++) {
+                                postings.nextPosition();
+                                postings.startOffset();
+                                postings.endOffset();
+                                postings.getPayload();
+                            }
+                        }
+                    }
+                }
+                final long proximityBytes = directory.getBytesRead() - postingsBytes - termBytes;
+                if (proximityBytes > 0) {
+                    stats.addProximity(field.name, proximityBytes);
+                }
+            }
+        }
+    }
+
+    void analyzePoints(SegmentReader reader, IndexDiskUsageStats stats) throws IOException {
+        PointsReader pointsReader = reader.getPointsReader();
+        if (pointsReader == null) {
+            return;
+        }
+        pointsReader = pointsReader.getMergeInstance();
+        final PointsVisitor visitor = new PointsVisitor(cancellationChecker);
+        for (FieldInfo field : reader.getFieldInfos()) {
+            cancellationChecker.checkForCancellation();
+            directory.resetAndTrackBytesRead(BytesReadTrackingMode.SIZE_ONLY);
+            if (field.getPointDimensionCount() > 0) {
+                final PointValues values = pointsReader.getValues(field.name);
+                values.intersect(visitor);
+                final long length = directory.getBytesRead();
+                if (length > 0) {
+                    stats.addPoints(field.name, length);
+                }
+            }
+        }
+    }
+
+    private static class PointsVisitor implements PointValues.IntersectVisitor {
+        private final CancellationChecker cancellationChecker;
+
+        PointsVisitor(CancellationChecker cancellationChecker) {
+            this.cancellationChecker = cancellationChecker;
+        }
+
+        @Override
+        public void visit(int docID) throws IOException {
+            assert false : "Must never be called";
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void visit(int docID, byte[] packedValue) throws IOException {
+            cancellationChecker.logEvent();
+            if (packedValue == null) {
+                throw new IllegalStateException("packedValue is null");
+            }
```

Comment:
Why do we need this check, this should be guaranteed by Lucene? Is it a leftover?

Reply:
Removed in https://github.com/elastic/elasticsearch/pull/74051/commits/dbfa965d3e80fdf505a42e0af0a52558b7e42cea.